contracts/contracts-foundry/src/attestation/AttestationAuther.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import {AccessControl} from "../../lib/openzeppelin-contracts/contracts/access/AccessControl.sol";
+import {IRiscZeroVerifier} from "../../lib/risc0-ethereum/contracts/src/IRiscZeroVerifier.sol";
+
+import {IAttestationVerifier} from "./IAttestationVerifier.sol";
+import {RiscZeroVerifier, RiscZeroVerifierDefault} from "./RiscZeroVerifier.sol";
+import {VerifiedKeys, VerifiedKeysDefault} from "./VerifiedKeys.sol";
+
+/// @title Attestation Auther Contract
+/// @notice Enclave key tracker for user contracts. Provides attestation verification, key management and image management.
+contract AttestationAuther is AccessControl, RiscZeroVerifierDefault, VerifiedKeysDefault {
+    /// @notice Role for approvers who can approve new images
+    bytes32 public constant APPROVER_ROLE = keccak256("APPROVER_ROLE");
+
+    /// @notice Role for revokers who can revoke existing images
+    bytes32 public constant REVOKER_ROLE = keccak256("REVOKER_ROLE");
+
+    IAttestationVerifier public attestationVerifier;
+
+    /// @dev Error thrown when attestation exceeds maximum allowed age
+    error AttestationAutherTooOld();
+
+    /// @dev Error thrown when public key has invalid format/length
+    error AttestationAutherPubkeyInvalid();
+
+    /// @notice Initializes the contract with roles and verification parameters
+    /// @param _admin Address to grant admin role
+    /// @param _approver Address to grant approver role
+    /// @param _revoker Address to grant revoker role
+    /// @param _attestationVerifier Address of attestation verifier contract
+    /// @param _verifier Address of RISC Zero verifier contract
+    /// @param _guestId Identifier for the zkVM guest program
+    /// @param _rootKey Initial root key for verification
+    /// @param _maxAgeMs Maximum age (in milliseconds) for valid attestations
+    /// @param _imageId Expected image ID for enclave verification
+    /// @param _family Image family
+    /// @dev Sets up role-based access control and initializes parent contracts
+    constructor(
+        address _admin,
+        address _approver,
+        address _revoker,
+        IAttestationVerifier _attestationVerifier,
+        IRiscZeroVerifier _verifier,
+        bytes32 _guestId,
+        bytes memory _rootKey,
+        uint256 _maxAgeMs,
+        bytes32 _imageId,
+        bytes32 _family
+    ) RiscZeroVerifier(_verifier, _guestId, _rootKey, _maxAgeMs) VerifiedKeys(_imageId, _family) {
```

Comment:
Max Age should probably be a parameter of `AttestationAuther` rather than `RiscZeroVerifier`

Reply:
it's meant to be used even without the auther